mpisppy/utils/amalgomator.py

```diff
@@ -242,29 +341,122 @@ def run(self):
                 self.best_inner_bound = results.Problem[0]['Upper bound']
                 self.best_outer_bound = results.Problem[0]['Lower bound']
             self.ef = ef
+            
+            if 'write_solution' in self.options:
+                if 'first_stage_solution' in self.options['write_solution']:
+                    sputils.write_ef_first_stage_solution(self.ef,
+                                                          self.options['write_solution']['first_stage_solution'])
+                if 'tree_solution' in self.options['write_solution']:
+                    sputils.write_ef_tree_solution(self.ef,
+                                                   self.options['write_solution']['tree_solution'])
+            
+            self.xhats = sputils.nonant_cache_from_ef(ef)
+            self.local_xhats = self.xhats #Every scenario is local for EF
+            self.first_stage_solution = {"ROOT": self.xhats["ROOT"]}
 
         else:
-            self.ef = None   # ???? do we want to retain these objects?
-            raise RuntimeError("We can only do EF right now")
+            self.ef = None
+            args = argparse.Namespace(**self.options)
+            
+            #Create a hub dict
+            hub_name = find_hub(self.options['cylinders'], self.is_multi)
+            hub_creator = getattr(vanilla, hub_name+'_hub')
+            beans = {"args":args,
+                           "scenario_creator": self.scenario_creator,
+                           "scenario_denouement": self.scenario_denouement,
+                           "all_scenario_names": self.scenario_names,
+                           "scenario_creator_kwargs": self.kwargs}
+            if self.is_multi:
+                    if "all_nodenames" in self.options :
+                        beans["all_nodenames"] = self.options["all_nodenames"]
+                    elif "branching_factors" in self.options:
+                        beans["branching_factors"] = self.options["branching_factors"]
+                    elif "BFs" in self.options:
+                        beans["branching_factors"] = self.options["BFs"]
```

Comment:
It might be better if this file just creates `all_nodenames` based on the options and passes that down into the hub/spoke creators, as opposed to passing the `branching_factors` to the hub/spoke creators. This may also help address the bug in aircond.

Further (probably not for this PR), I think we could just infer whether the problem is multistage or not by looking for `all_nodenames`, `branching_factors`, and `BFs` in options, and if none are there we assume the model to be 2-stage.